web/blog/tenzir-node-v4.27/index.md

```diff
@@ -0,0 +1,75 @@
+---
+title: "Tenzir Node v4.27: Amazon MSK IAM Integration"
+slug: tenzir-node-v4.27
+authors: [lava]
+date: 2025-01-29
+tags: [release, node]
+comments: true
+---
+
+Tenzir Node v4.27 enhances the charting capabilities and integrates with IAM for
+authenticating to Amazon MSK.
+
+![Tenzir Node v4.27](tenzir-node-v4.27.excalidraw.svg)
+
+[github-release]: https://github.com/tenzir/tenzir/releases/tag/v4.27.0
+
+<!-- truncate -->
+
+## AWS IAM Authentication for MSK
+
+The `load_kafka` and `save_kafka` operators can now authenticate with MSK using
+AWS IAM. To use IAM simply specify the `aws_iam` option with a record of
+configuration values.
+
+```tql
+load_kafka "kafkaesque-data", aws_iam={region: "eu-west-1"}
+```
+
+The above pipeline will try to fetch credentials in the following order:
+
+1. Checks your environment variables for AWS Credentials.
+2. Checks your `$HOME/.aws/credentials` file for a profile and credentials
+3. Contacts and logs in to a trusted identity provider. The login information to
+   these providers can either be on the environment variables: `AWS_ROLE_ARN`,
+`AWS_WEB_IDENTITY_TOKEN_FILE`, `AWS_ROLE_SESSION_NAME` or on a profile in your
+`$HOME/.aws/credentials`.
+4. Checks for an external method set as part of a profile on `$HOME/.aws/config`
+   to generate or look up credentials that isn't directly supported by AWS.
+5. Contacts the ECS Task Role to request credentials if Environment variable
+   `AWS_CONTAINER_CREDENTIALS_RELATIVE_URI` has been set.
+6. Contacts the EC2 Instance Metadata service to request credentials if
+   `AWS_EC2_METADATA_DISABLED` is NOT set to ON.
+
+### Assuming roles
+
+Roles can be assumed by giving the `assume_role` parameter to the the `aws_iam` option.
+
+```tql
+load_kafka "topic", aws_iam={region: "eu-west-1", assume_role: "<ROLE ARN>"}
+```
+
+The above pipeline attempts to fetch temporary credentials from Amazon STS.
+
+## Charts
```

Comment:
Let's add another section before charts with a warning banner (`:::warning`) that TQL1 is deprecated from now on, and a link to https://docs.tenzir.com/tql2-migration